plugin/pulpcore/plugin/download/asyncio/group.py

```diff
@@ -0,0 +1,190 @@
+import asyncio
+from collections import defaultdict
+from contextlib import suppress
+from itertools import chain
+
+from .http import HttpDownloader
+from .factory import DownloadFactory
+
+
+class GroupDownloader(object):
+    """
+    Download groups of files, with all downloads from all groups occurring in parallel.
+
+    This supports groups to be downloaded by adding them explicitly or via a iterator such as a
+    generator. The generator case is useful for limiting the number of in-memory objects.
+
+    Basic Usage:
+        >>> downloader = GroupDownloader(importer)
+        >>> artifact_a = DeferredArtifact(url=url_a md5='912ec803b2ce49e4a541068d495ab570')
```

Comment:
`DeferredArtifact` renamed to `RemoteArtifact`.

Reply:
Fixed in next push.